tests/e2e/performance/csi_tests/test_pvc_multi_snapshot_performance.py

```diff
@@ -1,42 +1,445 @@
+            Calculating the amount of storage which avaliable for the test
+            Creating namespace (project) for the test
+
+        """
+        log.info("Setting up the test environment")
+
+        super(TestPvcMultiSnapshotPerformance, self).setup()
+
+        # Getting the total Storage capacity
+        try:
+            self.ceph_capacity = int(self.ceph_cluster.get_ceph_capacity())
+        except Exception as err:
+            err_msg = f"Failed to get Storage capacity : {err}"
+            log.error(err_msg)
+            raise Exception(err_msg)
+
+        # Use 70% of the storage capacity in the test
+        self.capacity_to_use = int(self.ceph_capacity * 0.7)
+
+        # Creating new namespace for the test
+        self.nss_name = "pas-test-namespace"
+        log.info(f"Creating new namespace ({self.nss_name}) for the test")
+        try:
+            self.proj = helpers.create_project(project_name=self.nss_name)
+        except Exception:
+            log.error("Can not create new project")
+            raise CommandFailed(f"{self.nss_name} was not created")
+
+        # Initialize a general Snapshot object to use in the test
+        self.snapshot = OCP(kind="volumesnapshot", namespace=self.nss_name)
+
+    def teardown(self):
+        """
+        Cleaning up the environment :
+            Delete all snapshot
+            Delete the POD
+            Delete the PVC
+            Delete the StorageClass
+            Delete the VolumeSnapshotClass
+            Delete the data pool
+            Switch to the default namespace
+            Delete the tested namespace
+
+        """
+        log.info("Cleanup the test environment")
+
+        # Deleting al snapshots from the cluster
+        try:
+            snapshot_list = self.snapshot.get(all_namespaces=True)["items"]
+        except Exception as err:
+            log.error(f"Cannot get the list of snapshots : {err}")
+            snapshot_list = []
+
+        log.info(f"Trying to delete all ({len(snapshot_list)}) Snapshots")
+        log.debug(
+            f"The list of all snapshots is : {json.dumps(snapshot_list, indent=3)}"
+        )
+        for vs in snapshot_list:
+            snap_name = vs["metadata"]["name"]
+            log.info(f"Try to delete {snap_name}")
+            try:
+                self.snapshot.delete(resource_name=snap_name)
+            except Exception as err:
+                log.error(f"Can not delete {snap_name} : {err}")
+
+        # Deleting the pod which wrote data to the pvc
+        log.info(f"Deleting the test POD : {self.pod_obj.name}")
+        try:
+            self.pod_obj.delete()
+        except Exception as ex:
+            log.error(f"Can not delete the test pod : {ex}")
+
+        # Deleting the PVC which used in the test.
+        log.info(f"Delete the PVC : {self.pvc_obj.name}")
+        try:
+            self.pvc_obj.delete()
+        except Exception as ex:
+            log.error(f"Can not delete the test pvc : {ex}")
```

Comment:
It's good to add some validation to confirm Snapshots --> PODs --> PVCs are deleted on each step.

Reply:
done